sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -299,7 +308,7 @@ class Column(val expr: Expression) extends Logging {
    * @group expr_ops
    * @since 2.0.0
     */
-  def =!= (other: Any): Column = withExpr{ Not(EqualTo(expr, lit(other).expr)) }
+  def =!= (other: Any): Column = withBinaryCompareExpr { Not(EqualTo(expr, lit(other).expr)) }
```

Comment:
Except `===`, the others are not meaningless. Users might use it for return an empty date set. For example, `x =!= x`